upstream-sha256: 3632760fd940fc0035f35d8b236d4dd3a589a056231b9357205344981167608b
actual-sha256: 57baccfae0a9d28a9d3f30cd28a535e273182c42117d1f64914af17420d90f0f
--- upstream.expected
+++ tolk.actual
@@ -3,120 +3,123 @@
    2 Ops.CAST            :            dtypes.int                               ['16']                           dtypes.int
    3 Ops.RANGE           : 1          dtypes.int                               [2]                              (1, AxisType.WEAK)
    4 Ops.INDEX           : 1          dtypes.float                             [0, 3]                           None
-   5 Ops.PARAM           :            dtypes.float                             []                               ParamArg(1, dtypes.float, 1024, device='CPU')
-   6 Ops.CONST           :            dtypes.weakint                           []                               5
-   7 Ops.CAST            :            dtypes.int                               ['5']                            dtypes.int
-   8 Ops.SHL             : 1          dtypes.int                               [3, 7]                           None
-   9 Ops.CONST           :            dtypes.weakint                           []                               4
-  10 Ops.CAST            :            dtypes.int                               ['4']                            dtypes.int
-  11 Ops.SHRINK          : 1          dtypes.float                             [5, 8, 10]                       None
-  12 Ops.LOAD            : 1          dtypes.float                             [11]                             None
-  13 Ops.CONST           :            dtypes.weakint                           []                               0
-  14 Ops.CAST            :            dtypes.int                               ['0']                            dtypes.int
-  15 Ops.INDEX           : 1          dtypes.float                             [12, 14]                         None
-  16 Ops.CONST           :            dtypes.weakint                           []                               1
-  17 Ops.CAST            :            dtypes.int                               ['1']                            dtypes.int
-  18 Ops.INDEX           : 1          dtypes.float                             [12, 17]                         None
-  19 Ops.ADD             : 1          dtypes.float                             [15, 18]                         None
-  20 Ops.CONST           :            dtypes.weakint                           []                               2
-  21 Ops.CAST            :            dtypes.int                               ['2']                            dtypes.int
-  22 Ops.INDEX           : 1          dtypes.float                             [12, 21]                         None
-  23 Ops.ADD             : 1          dtypes.float                             [19, 22]                         None
-  24 Ops.CONST           :            dtypes.weakint                           []                               3
-  25 Ops.CAST            :            dtypes.int                               ['3']                            dtypes.int
-  26 Ops.INDEX           : 1          dtypes.float                             [12, 25]                         None
-  27 Ops.ADD             : 1          dtypes.float                             [23, 26]                         None
-  28 Ops.ADD             : 1          dtypes.int                               [8, 10]                          None
-  29 Ops.SHRINK          : 1          dtypes.float                             [5, 28, 10]                      None
-  30 Ops.LOAD            : 1          dtypes.float                             [29]                             None
-  31 Ops.INDEX           : 1          dtypes.float                             [30, 14]                         None
-  32 Ops.ADD             : 1          dtypes.float                             [27, 31]                         None
-  33 Ops.INDEX           : 1          dtypes.float                             [30, 17]                         None
-  34 Ops.ADD             : 1          dtypes.float                             [32, 33]                         None
-  35 Ops.INDEX           : 1          dtypes.float                             [30, 21]                         None
-  36 Ops.ADD             : 1          dtypes.float                             [34, 35]                         None
-  37 Ops.INDEX           : 1          dtypes.float                             [30, 25]                         None
-  38 Ops.ADD             : 1          dtypes.float                             [36, 37]                         None
-  39 Ops.CONST           :            dtypes.weakint                           []                               8
-  40 Ops.CAST            :            dtypes.int                               ['8']                            dtypes.int
-  41 Ops.ADD             : 1          dtypes.int                               [8, 40]                          None
-  42 Ops.SHRINK          : 1          dtypes.float                             [5, 41, 10]                      None
-  43 Ops.LOAD            : 1          dtypes.float                             [42]                             None
-  44 Ops.INDEX           : 1          dtypes.float                             [43, 14]                         None
-  45 Ops.ADD             : 1          dtypes.float                             [38, 44]                         None
-  46 Ops.INDEX           : 1          dtypes.float                             [43, 17]                         None
-  47 Ops.ADD             : 1          dtypes.float                             [45, 46]                         None
-  48 Ops.INDEX           : 1          dtypes.float                             [43, 21]                         None
-  49 Ops.ADD             : 1          dtypes.float                             [47, 48]                         None
-  50 Ops.INDEX           : 1          dtypes.float                             [43, 25]                         None
-  51 Ops.ADD             : 1          dtypes.float                             [49, 50]                         None
-  52 Ops.CONST           :            dtypes.weakint                           []                               12
-  53 Ops.CAST            :            dtypes.int                               ['12']                           dtypes.int
-  54 Ops.ADD             : 1          dtypes.int                               [8, 53]                          None
-  55 Ops.SHRINK          : 1          dtypes.float                             [5, 54, 10]                      None
-  56 Ops.LOAD            : 1          dtypes.float                             [55]                             None
-  57 Ops.INDEX           : 1          dtypes.float                             [56, 14]                         None
-  58 Ops.ADD             : 1          dtypes.float                             [51, 57]                         None
-  59 Ops.INDEX           : 1          dtypes.float                             [56, 17]                         None
-  60 Ops.ADD             : 1          dtypes.float                             [58, 59]                         None
-  61 Ops.INDEX           : 1          dtypes.float                             [56, 21]                         None
-  62 Ops.ADD             : 1          dtypes.float                             [60, 61]                         None
-  63 Ops.INDEX           : 1          dtypes.float                             [56, 25]                         None
-  64 Ops.ADD             : 1          dtypes.float                             [62, 63]                         None
-  65 Ops.ADD             : 1          dtypes.int                               [8, 2]                           None
-  66 Ops.SHRINK          : 1          dtypes.float                             [5, 65, 10]                      None
-  67 Ops.LOAD            : 1          dtypes.float                             [66]                             None
-  68 Ops.INDEX           : 1          dtypes.float                             [67, 14]                         None
-  69 Ops.ADD             : 1          dtypes.float                             [64, 68]                         None
-  70 Ops.INDEX           : 1          dtypes.float                             [67, 17]                         None
-  71 Ops.ADD             : 1          dtypes.float                             [69, 70]                         None
-  72 Ops.INDEX           : 1          dtypes.float                             [67, 21]                         None
-  73 Ops.ADD             : 1          dtypes.float                             [71, 72]                         None
-  74 Ops.INDEX           : 1          dtypes.float                             [67, 25]                         None
-  75 Ops.ADD             : 1          dtypes.float                             [73, 74]                         None
-  76 Ops.CONST           :            dtypes.weakint                           []                               20
-  77 Ops.CAST            :            dtypes.int                               ['20']                           dtypes.int
-  78 Ops.ADD             : 1          dtypes.int                               [8, 77]                          None
-  79 Ops.SHRINK          : 1          dtypes.float                             [5, 78, 10]                      None
-  80 Ops.LOAD            : 1          dtypes.float                             [79]                             None
-  81 Ops.INDEX           : 1          dtypes.float                             [80, 14]                         None
-  82 Ops.ADD             : 1          dtypes.float                             [75, 81]                         None
-  83 Ops.INDEX           : 1          dtypes.float                             [80, 17]                         None
-  84 Ops.ADD             : 1          dtypes.float                             [82, 83]                         None
-  85 Ops.INDEX           : 1          dtypes.float                             [80, 21]                         None
-  86 Ops.ADD             : 1          dtypes.float                             [84, 85]                         None
-  87 Ops.INDEX           : 1          dtypes.float                             [80, 25]                         None
-  88 Ops.ADD             : 1          dtypes.float                             [86, 87]                         None
-  89 Ops.CONST           :            dtypes.weakint                           []                               24
-  90 Ops.CAST            :            dtypes.int                               ['24']                           dtypes.int
-  91 Ops.ADD             : 1          dtypes.int                               [8, 90]                          None
-  92 Ops.SHRINK          : 1          dtypes.float                             [5, 91, 10]                      None
-  93 Ops.LOAD            : 1          dtypes.float                             [92]                             None
-  94 Ops.INDEX           : 1          dtypes.float                             [93, 14]                         None
-  95 Ops.ADD             : 1          dtypes.float                             [88, 94]                         None
-  96 Ops.INDEX           : 1          dtypes.float                             [93, 17]                         None
-  97 Ops.ADD             : 1          dtypes.float                             [95, 96]                         None
-  98 Ops.INDEX           : 1          dtypes.float                             [93, 21]                         None
-  99 Ops.ADD             : 1          dtypes.float                             [97, 98]                         None
- 100 Ops.INDEX           : 1          dtypes.float                             [93, 25]                         None
- 101 Ops.ADD             : 1          dtypes.float                             [99, 100]                        None
- 102 Ops.CONST           :            dtypes.weakint                           []                               28
- 103 Ops.CAST            :            dtypes.int                               ['28']                           dtypes.int
- 104 Ops.ADD             : 1          dtypes.int                               [8, 103]                         None
- 105 Ops.SHRINK          : 1          dtypes.float                             [5, 104, 10]                     None
- 106 Ops.LOAD            : 1          dtypes.float                             [105]                            None
- 107 Ops.INDEX           : 1          dtypes.float                             [106, 14]                        None
- 108 Ops.ADD             : 1          dtypes.float                             [101, 107]                       None
- 109 Ops.INDEX           : 1          dtypes.float                             [106, 17]                        None
- 110 Ops.ADD             : 1          dtypes.float                             [108, 109]                       None
- 111 Ops.INDEX           : 1          dtypes.float                             [106, 21]                        None
- 112 Ops.ADD             : 1          dtypes.float                             [110, 111]                       None
- 113 Ops.INDEX           : 1          dtypes.float                             [106, 25]                        None
- 114 Ops.ADD             : 1          dtypes.float                             [112, 113]                       None
- 115 Ops.PARAM           :            dtypes.float                             []                               ParamArg(2, dtypes.float, 16, device='CPU')
- 116 Ops.INDEX           : 1          dtypes.float                             [115, 3]                         None
- 117 Ops.LOAD            : 1          dtypes.float                             [116]                            None
- 118 Ops.ADD             : 1          dtypes.float                             [114, 117]                       None
- 119 Ops.STORE           : 1          dtypes.void                              [4, 118]                         None
- 120 Ops.END             :            dtypes.void                              [119, 3]                         None
- 121 Ops.SINK            :            dtypes.void                              [120]                            KernelInfo(name='r_16_32', applied_opts=(Opt(op=OptOps.SPLIT, axis=1, arg=(0, AxisType.UNROLL)),), opts_to_apply=None, estimates=None, beam=0)
+   5 Ops.CONST           :            dtypes.weakfloat                         []                               0.0
+   6 Ops.CAST            :            dtypes.float                             ['0.0']                          dtypes.float
+   7 Ops.PARAM           :            dtypes.float                             []                               ParamArg(1, dtypes.float, 1024, device='CPU')
+   8 Ops.CONST           :            dtypes.weakint                           []                               5
+   9 Ops.CAST            :            dtypes.int                               ['5']                            dtypes.int
+  10 Ops.SHL             : 1          dtypes.int                               [3, 9]                           None
+  11 Ops.CONST           :            dtypes.weakint                           []                               4
+  12 Ops.CAST            :            dtypes.int                               ['4']                            dtypes.int
+  13 Ops.SHRINK          : 1          dtypes.float                             [7, 10, 12]                      None
+  14 Ops.LOAD            : 1          dtypes.float                             [13]                             None
+  15 Ops.CONST           :            dtypes.weakint                           []                               0
+  16 Ops.CAST            :            dtypes.int                               ['0']                            dtypes.int
+  17 Ops.INDEX           : 1          dtypes.float                             [14, 16]                         None
+  18 Ops.ADD             : 1          dtypes.float                             [6, 17]                          None
+  19 Ops.CONST           :            dtypes.weakint                           []                               1
+  20 Ops.CAST            :            dtypes.int                               ['1']                            dtypes.int
+  21 Ops.INDEX           : 1          dtypes.float                             [14, 20]                         None
+  22 Ops.ADD             : 1          dtypes.float                             [18, 21]                         None
+  23 Ops.CONST           :            dtypes.weakint                           []                               2
+  24 Ops.CAST            :            dtypes.int                               ['2']                            dtypes.int
+  25 Ops.INDEX           : 1          dtypes.float                             [14, 24]                         None
+  26 Ops.ADD             : 1          dtypes.float                             [22, 25]                         None
+  27 Ops.CONST           :            dtypes.weakint                           []                               3
+  28 Ops.CAST            :            dtypes.int                               ['3']                            dtypes.int
+  29 Ops.INDEX           : 1          dtypes.float                             [14, 28]                         None
+  30 Ops.ADD             : 1          dtypes.float                             [26, 29]                         None
+  31 Ops.ADD             : 1          dtypes.int                               [10, 12]                         None
+  32 Ops.SHRINK          : 1          dtypes.float                             [7, 31, 12]                      None
+  33 Ops.LOAD            : 1          dtypes.float                             [32]                             None
+  34 Ops.INDEX           : 1          dtypes.float                             [33, 16]                         None
+  35 Ops.ADD             : 1          dtypes.float                             [30, 34]                         None
+  36 Ops.INDEX           : 1          dtypes.float                             [33, 20]                         None
+  37 Ops.ADD             : 1          dtypes.float                             [35, 36]                         None
+  38 Ops.INDEX           : 1          dtypes.float                             [33, 24]                         None
+  39 Ops.ADD             : 1          dtypes.float                             [37, 38]                         None
+  40 Ops.INDEX           : 1          dtypes.float                             [33, 28]                         None
+  41 Ops.ADD             : 1          dtypes.float                             [39, 40]                         None
+  42 Ops.CONST           :            dtypes.weakint                           []                               8
+  43 Ops.CAST            :            dtypes.int                               ['8']                            dtypes.int
+  44 Ops.ADD             : 1          dtypes.int                               [10, 43]                         None
+  45 Ops.SHRINK          : 1          dtypes.float                             [7, 44, 12]                      None
+  46 Ops.LOAD            : 1          dtypes.float                             [45]                             None
+  47 Ops.INDEX           : 1          dtypes.float                             [46, 16]                         None
+  48 Ops.ADD             : 1          dtypes.float                             [41, 47]                         None
+  49 Ops.INDEX           : 1          dtypes.float                             [46, 20]                         None
+  50 Ops.ADD             : 1          dtypes.float                             [48, 49]                         None
+  51 Ops.INDEX           : 1          dtypes.float                             [46, 24]                         None
+  52 Ops.ADD             : 1          dtypes.float                             [50, 51]                         None
+  53 Ops.INDEX           : 1          dtypes.float                             [46, 28]                         None
+  54 Ops.ADD             : 1          dtypes.float                             [52, 53]                         None
+  55 Ops.CONST           :            dtypes.weakint                           []                               12
+  56 Ops.CAST            :            dtypes.int                               ['12']                           dtypes.int
+  57 Ops.ADD             : 1          dtypes.int                               [10, 56]                         None
+  58 Ops.SHRINK          : 1          dtypes.float                             [7, 57, 12]                      None
+  59 Ops.LOAD            : 1          dtypes.float                             [58]                             None
+  60 Ops.INDEX           : 1          dtypes.float                             [59, 16]                         None
+  61 Ops.ADD             : 1          dtypes.float                             [54, 60]                         None
+  62 Ops.INDEX           : 1          dtypes.float                             [59, 20]                         None
+  63 Ops.ADD             : 1          dtypes.float                             [61, 62]                         None
+  64 Ops.INDEX           : 1          dtypes.float                             [59, 24]                         None
+  65 Ops.ADD             : 1          dtypes.float                             [63, 64]                         None
+  66 Ops.INDEX           : 1          dtypes.float                             [59, 28]                         None
+  67 Ops.ADD             : 1          dtypes.float                             [65, 66]                         None
+  68 Ops.ADD             : 1          dtypes.int                               [10, 2]                          None
+  69 Ops.SHRINK          : 1          dtypes.float                             [7, 68, 12]                      None
+  70 Ops.LOAD            : 1          dtypes.float                             [69]                             None
+  71 Ops.INDEX           : 1          dtypes.float                             [70, 16]                         None
+  72 Ops.ADD             : 1          dtypes.float                             [67, 71]                         None
+  73 Ops.INDEX           : 1          dtypes.float                             [70, 20]                         None
+  74 Ops.ADD             : 1          dtypes.float                             [72, 73]                         None
+  75 Ops.INDEX           : 1          dtypes.float                             [70, 24]                         None
+  76 Ops.ADD             : 1          dtypes.float                             [74, 75]                         None
+  77 Ops.INDEX           : 1          dtypes.float                             [70, 28]                         None
+  78 Ops.ADD             : 1          dtypes.float                             [76, 77]                         None
+  79 Ops.CONST           :            dtypes.weakint                           []                               20
+  80 Ops.CAST            :            dtypes.int                               ['20']                           dtypes.int
+  81 Ops.ADD             : 1          dtypes.int                               [10, 80]                         None
+  82 Ops.SHRINK          : 1          dtypes.float                             [7, 81, 12]                      None
+  83 Ops.LOAD            : 1          dtypes.float                             [82]                             None
+  84 Ops.INDEX           : 1          dtypes.float                             [83, 16]                         None
+  85 Ops.ADD             : 1          dtypes.float                             [78, 84]                         None
+  86 Ops.INDEX           : 1          dtypes.float                             [83, 20]                         None
+  87 Ops.ADD             : 1          dtypes.float                             [85, 86]                         None
+  88 Ops.INDEX           : 1          dtypes.float                             [83, 24]                         None
+  89 Ops.ADD             : 1          dtypes.float                             [87, 88]                         None
+  90 Ops.INDEX           : 1          dtypes.float                             [83, 28]                         None
+  91 Ops.ADD             : 1          dtypes.float                             [89, 90]                         None
+  92 Ops.CONST           :            dtypes.weakint                           []                               24
+  93 Ops.CAST            :            dtypes.int                               ['24']                           dtypes.int
+  94 Ops.ADD             : 1          dtypes.int                               [10, 93]                         None
+  95 Ops.SHRINK          : 1          dtypes.float                             [7, 94, 12]                      None
+  96 Ops.LOAD            : 1          dtypes.float                             [95]                             None
+  97 Ops.INDEX           : 1          dtypes.float                             [96, 16]                         None
+  98 Ops.ADD             : 1          dtypes.float                             [91, 97]                         None
+  99 Ops.INDEX           : 1          dtypes.float                             [96, 20]                         None
+ 100 Ops.ADD             : 1          dtypes.float                             [98, 99]                         None
+ 101 Ops.INDEX           : 1          dtypes.float                             [96, 24]                         None
+ 102 Ops.ADD             : 1          dtypes.float                             [100, 101]                       None
+ 103 Ops.INDEX           : 1          dtypes.float                             [96, 28]                         None
+ 104 Ops.ADD             : 1          dtypes.float                             [102, 103]                       None
+ 105 Ops.CONST           :            dtypes.weakint                           []                               28
+ 106 Ops.CAST            :            dtypes.int                               ['28']                           dtypes.int
+ 107 Ops.ADD             : 1          dtypes.int                               [10, 106]                        None
+ 108 Ops.SHRINK          : 1          dtypes.float                             [7, 107, 12]                     None
+ 109 Ops.LOAD            : 1          dtypes.float                             [108]                            None
+ 110 Ops.INDEX           : 1          dtypes.float                             [109, 16]                        None
+ 111 Ops.ADD             : 1          dtypes.float                             [104, 110]                       None
+ 112 Ops.INDEX           : 1          dtypes.float                             [109, 20]                        None
+ 113 Ops.ADD             : 1          dtypes.float                             [111, 112]                       None
+ 114 Ops.INDEX           : 1          dtypes.float                             [109, 24]                        None
+ 115 Ops.ADD             : 1          dtypes.float                             [113, 114]                       None
+ 116 Ops.INDEX           : 1          dtypes.float                             [109, 28]                        None
+ 117 Ops.ADD             : 1          dtypes.float                             [115, 116]                       None
+ 118 Ops.PARAM           :            dtypes.float                             []                               ParamArg(2, dtypes.float, 16, device='CPU')
+ 119 Ops.INDEX           : 1          dtypes.float                             [118, 3]                         None
+ 120 Ops.LOAD            : 1          dtypes.float                             [119]                            None
+ 121 Ops.ADD             : 1          dtypes.float                             [117, 120]                       None
+ 122 Ops.STORE           : 1          dtypes.void                              [4, 121]                         None
+ 123 Ops.END             :            dtypes.void                              [122, 3]                         None
+ 124 Ops.SINK            :            dtypes.void                              [123]                            KernelInfo(name='r_16_32', applied_opts=(Opt(op=OptOps.SPLIT, axis=1, arg=(0, AxisType.UNROLL)),), opts_to_apply=None, estimates=None, beam=0)
